lib/actions/appleMusic.js

```diff
@@ -13,46 +16,177 @@ function getAlbumUri(id) {
+                return track.artistName.toLowerCase();
+              }).sort();
+              var tracks = tracksJson.results.map(function(track) {
+                return track.trackName.toLowerCase();
+              }).sort();
+
+              // Determine if the search results are many tracks for an artist OR a specific song title.
+              var prevArtist=artists[0];
+              var prevTrack=tracks[0];
+  
+              for (var i=1; i < tracksJson.resultCount;i++) {
+                if (artists[i] != prevArtist) {
+                  artistCount++;
+                  prevArtist = artists[i];
+                }
+                if (tracks[i] != prevTrack) {
+                  trackCount++;
+                  prevTrack = tracks[i];
+                }  
+              }
+          
+              // If there are at least twice as many track differences that artist differences, then load all of the artist's tracks into the queue, ELSE play just the first track now (iTunes returns the most popular track first) 
+              if (trackCount/artistCount > 2) {
+                const queueURI = 'x-rincon-queue:' + player.uuid + '#0';
+                
+                // Filtered list of tracks to play
+                var queueTracks = tracksJson.results.reduce(function(tracksArray, track) {
+                  if (track.isStreamable) {
+                    var skip = false;
+  
+                    for (var j=0; (j < tracksArray.length) && !skip ; j++) {
+                      // Skip duplicate songs
+                      skip = (track.trackName == tracksArray[j]);
+                    }
+                    
+                    if (!skip) {
+                      var metadataID = METADATA_URI_STARTERS[type] + encodeURIComponent('song:' + track.trackId);
+                      var metadata = getMetadata(metadataID, PARENTS[type], CLASSES[type],'');
+                      var uri = uriTemplates[type](encodeURIComponent('song:' + track.trackId));
+
+                      tracksArray.push({trackName:track.trackName, uri:uri, metadata:metadata});
+                    }
+                  } 
+                  return tracksArray;
+                }, []);
+
+                nextTrackNo = 0;
+                return player.coordinator.clearQueue()
+                  .then(() => player.coordinator.setAVTransport(queueURI, ''))
+                  .then(() => player.coordinator.addURIToQueue(queueTracks[0].uri, queueTracks[0].metadata, true, 1))
+                  .then(() => player.coordinator.play())
+                  .then(() => {
+                    return queueTracks.reduce(function(promise, track){
+                      if (++nextTrackNo > 1) {
+                        return promise.then(() => player.coordinator.addURIToQueue(track.uri, track.metadata, true, nextTrackNo));
+                      } else {
+                        return promise.then(() => {});
+                      }   
+                    }, Promise.resolve());
```

Comment:
Suggestion: You can do like this instead:
```
return player.coordinator.clearQueue()
   .then(() => player.coordinator.setAVTransport(queueURI, ''))
   .then(() => {
     return queueTracks.reduce(function(promise, track, index) {
       promise = promise.then(() => player.coordinator.addURIToQueue(track.uri, track.metadata, true, index + 1)); // not sure if +1 is required, it will start on 0 at least.
       if (index == 0) {
         promise = promise.then(() => player.play());
       }
      return promise;
     }, Promise.resolve());
```